appGeo/src/tests/game.spec.ts

```diff
@@ -1,325 +1,430 @@
+QUnit.test("Game Start and endGame", function(assert) {
+    const now = new Date();
+    const next_week = new Date(now.getTime() + 7 * 24 * 60 * 60 * 1000);
 
-//     assert_equal(test_game.end(),FAILURE, "Unstarted game cannot end");
-//     assert_false(test_game.isGameActive(), "Unstarted game is not active");
+    const test_map = new CampusMap();
 
-//     assert_equal(test_game.start(), SUCCESS, "Game can start with five or more players");
-//     assert_true(test_game.isGameActive(), "Started game is active");
+    const Location1 = new Location(1);
 
-//     assert_equal(test_game.getPlayerCount(), 5, "Can count players in game");
-//     assert_equal(test_game.getRoleCount(CIVILIAN), 3, "Should start game with 3 civilians");
-//     assert_equal(test_game.getRoleCount(KILLER), 1, "Should start game with 1 killer");
+    const player1 = new Player(1, 'player1', Location1, ALIVE); 
+    const player2 = new Player(2, 'player2', Location1, ALIVE);
+    const player3 = new Player(3, 'player3', Location1, ALIVE);
 
-//     assert_equal(test_game.getFractionRole(CIVILIAN), 0.8, "Should be 80% civilians");
-//     assert_equal(test_game.getFractionRole(KILLER), 0.2, "Should be 20% killers");
+    const playerArray = [player1, player2, player3];
+    const player_map = new Map(
+        playerArray.map(object => {
+          return [object.getUserID(), object];
+        }),
+      );
 
-//     assert_equal(test_game.end(), SUCCESS, "Can stop a game");
-//     assert_false(test_game.isGameActive(), "Stopped game is not active");
+    const test_game = new Game(next_week, test_map, player_map);
 
-//     playerArray.push(player4);
-//     playerArray.push(player5);
+    assert.equal(test_game.startGame(), SUCCESS, "Game starts with players");
+    assert.equal(test_game.getGameActive(), ACTIVE, "Started game is not active");
+    assert.equal(test_game.endGame(),SUCCESS, "Started game cannot endGame");
+    assert.equal(test_game.getGameActive(), INACTIVE, "Unstarted game is not active");
 
-//     const soon = new Date(now.getTime() + 1000);
+    const player4 = new Player(4, 'player4', Location1, ALIVE);
+    const player5 = new Player(5, 'player5', Location1, ALIVE);
 
-//     const test_game2 = new Game(playerArray, next_week, test_map);
-//     assert_equal(test_game.start(), SUCCESS, "Game can start with five or more players");
+    test_game.addPlayer(player4);
+    test_game.addPlayer(player5);
 
-//     //should delay by 1 second, need to write this in
+    assert.equal(test_game.getPlayerCount(), 5, "Can count players outside of game");
+    assert.equal(test_game.getRoleCount(CIVILIAN), 0, "No role assigned, no civilians");
+    assert.equal(test_game.getRoleCount(KILLER), 0, "No role assigned, no killers");
 
-//     assert_false(test_game.isGameActive(), "Stopped game is not active");
+    // assert.equal(test_game.getFractionRole(CIVILIAN), 0, "No role assigned, no civilians");
```

Comment:
Please put a comment on why this is commented out or delete it. Otherwise we won't know why it's commented out later.